app/main/views/g_cloud.py

```diff
@@ -638,9 +638,27 @@ def which_service_won_contract(framework_family, project_id):
 
     form = WhichServiceWonTheContractForm(services)
 
-    if request.method == "POST" and form.validate_on_submit():
-        flash('Contract awarded.')
-        return redirect(url_for('.tell_us_about_contract', framework_family=framework_family, project_id=project['id']))
+    if form.validate_on_submit():
+        try:
+            outcome = data_api_client.create_direct_award_project_outcome_award(
+                project_id=project_id,
+                awarded_service_id=form.which_service_won_the_contract.data,
+                user_email=current_user.email_address)
+        except HTTPError:
+            flash(f"There was an error updating '{project['name']}', please try again later", 'error')
+            return render_template(
+                'direct-award/which-service-won-contract.html',
+                project=project,
+                errors={},
+                framework=framework,
+                services=services,
+                form=form,
+            ), 400
```

Comment:
So I think if we're doing this, we should proxy the error code from the API back to the client. It's not correct to give the client a 400 if the API gave us a 500!

However I think if we're aiming to fully validate the form before passing the data to the backend, then we don't need to catch errors here at all.

The specific error scenarios that the API can detect that the frontend cannot, like if you already set the outcome in another tab whilst leaving this form open in another, are often not helped by asking the user to try again later with the same form again!

If decide we must catch some `HTTPError`s like this then I think we should at least be re-raising for 5xx errors, so when diagnosing API errors we can see the full trace from the calling apps too.

Reply:
I will leave them out